components/Ripple/src/MDCRippleTouchController.m

```diff
@@ -30,13 +38,25 @@ - (instancetype)initWithView:(UIView *)view {
     _gestureRecognizer.delaysTouchesEnded = NO;
 
     _shouldProcessRippleWithScrollViewGestures = YES;
-    _view = view;
-    [_view addGestureRecognizer:_gestureRecognizer];
+    _rippleView = [[MDCRippleView alloc] init];
+  }
+  return self;
+}
+
+- (void)addRippleToView:(UIView *)view {
+  [self configureRippleWithView:view];
+}
+
+- (void)configureRippleWithView:(UIView *)view {
+  _view = view;
+  [_view addGestureRecognizer:_gestureRecognizer];
```

Comment:
Do you need to remove the gesture recognizer from the old view before you swap them out?

```objc
[_view removeGestureRecognizer:_gestureRecognizer];
_view = view;
[_view addGestureRecognizer:_gestureRecognizer];
```